addons/save-in-project-page/userscript.js

```diff
@@ -0,0 +1,21 @@
+import downloadBlob from "../../libraries/common/cs/download-blob.js";
+
+export default async function ({ addon, console }) {
+  const vm = addon.tab.traps.vm;
+
+  async function download() {
+    const project = await vm.saveProjectSb3();
+    const title = await addon.tab.waitForElement(".project-title");
+    downloadBlob(`${title.innerHTML}.sb3`, project);
```

Comment:
`addon.tab.redux.state.scratchGui.projectTitle` works, but seemingly only when you're in the editor.

Reply:
innerText seems good enough as a default name of the file

Reply:
> innerText seems good enough as a default name of the file

Yeah, it's not like we've done this before, have we?

> what is wrong with using innertext? it works fine for me?

As I [said](https://github.com/ScratchAddons/ScratchAddons/pull/6931#discussion_r1417983667)...

> innerHTML returns a string of the HTML for the project title input box when you have created the project. innerText and textContent both return an empty string in the same case.

Let me show you:

![innerHTML of read-only project title](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/f9f06b11-a3a0-4b62-b067-4d26de01eafb)
![innerHTML of editable project title](https://github.com/ScratchAddons/ScratchAddons/assets/106490990/418eee03-979d-4bdd-b096-fedd8a6922f9)

Reply:
oh, my bad I didnt realise it was an input, well is there an API for checking if you are the project creator? because to get around this ill just get the value of the input instead

Reply:
fixed by [commit](https://github.com/ScratchAddons/ScratchAddons/pull/6931/commits/d60ac75a2484524d00d93ec38f1cc0761d3ecb3f)